ansible_base/api_documentation/postprocessing_hooks.py

```diff
@@ -0,0 +1,435 @@
+import logging
+
+from inflection import singularize
+
+from ansible_base.api_documentation.preprocessing_hooks import OPERATION_CLASS_MAP, RESOURCE_PURPOSE_MAP, SKIP_AI_DESCRIPTION_PREFIXES
+from ansible_base.lib.utils.api_path_utils import (
+    extract_operation_action,
+    extract_operation_prefix,
+    filter_api_prefixes,
+    parse_path_segments,
+)
+
+logger = logging.getLogger('ansible_base.api_documentation.postprocessing_hooks')
+
+# Valid HTTP methods for OpenAPI operations
+HTTP_METHODS = ['get', 'post', 'put', 'patch', 'delete', 'head', 'options']
+
+# Operation name mappings for different HTTP methods and actions
+OPERATION_DESCRIPTIVE_NAMES = {
+    'list': 'List all',
+    'create': 'Create new',
+    'retrieve': 'Retrieve single',
+    'read': 'Retrieve single',
+    'update': 'Update existing',
+    'partial_update': 'Partially update existing',
+    'destroy': 'Delete existing',
+    'delete': 'Delete existing',
+}
+
+# Operation description templates for generating descriptions from resource_purpose
+# These lambdas are used to create contextual descriptions for CRUD operations
+OPERATION_DESCRIPTION_TEMPLATES = {
+    'list': lambda p: f"List {p}",
+    'retrieve': lambda p: f"Retrieve a {singularize_resource_purpose(p)}",
+    'read': lambda p: f"Retrieve a {singularize_resource_purpose(p)}",
+    'create': lambda p: f"Create a {singularize_resource_purpose(p)}",
+    'update': lambda p: f"Update a {singularize_resource_purpose(p)}",
+    'partial_update': lambda p: f"Update a {singularize_resource_purpose(p)}",
+    'destroy': lambda p: f"Delete a {singularize_resource_purpose(p)}",
+    'delete': lambda p: f"Delete a {singularize_resource_purpose(p)}",
+}
+
+
+def _extract_resource_parts_from_operation_id(operation_id, action):
```

Comment:
I would /strongly/ suggest mypy references everywhere if possible.

Reply:
Every method should now have mypy types. Note that for `add_x_ai_description` I'm using `Any` to prevent importing libraries.